File: EmbyStat.Web/ClientApp/src/app/shared/services/job.service.ts
import { Observable } from 'rxjs';

import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';

import { Job } from '../models/jobs/job';

@Injectable({
  providedIn: 'root'
})
export class JobService {
  private readonly baseUrl = '/api/job';
  private fireUrl = `${this.baseUrl}/fire/`;
  private getMediaSyncJobUrl = `${this.baseUrl}/mediasync`;

  constructor(private http: HttpClient) {

  }

  getAll(): Observable<Job[]> {
    return this.http.get<Job[]>(this.baseUrl);
  }

  getById(id: string): Observable<Job> {
    return this.http.get<Job>(this.baseUrl + '/' + id);
  }

  updateTrigger(id: string, cron: string): Observable<void> {
    const httpParams = new HttpParams().append('cron', cron);
    return this.http.patch<void>(this.baseUrl + '/' + id, {}, { params: httpParams });
  }

  fireJob(id: string): Observable<void> {
    return this.http.post<void>(this.fireUrl + id, {});
  }

  getMediaSyncJob(): Observable<Job> {
    return this.http.get<Job>(this.getMediaSyncJobUrl);
  }
}

File: EmbyStat.Core/Jobs/JobService.cs
using EmbyStat.Common.Models.Entities;
using EmbyStat.Core.Jobs.Interfaces;

namespace EmbyStat.Core.Jobs;

public class JobService : IJobService
{
    private readonly IJobRepository _jobRepository;

    public JobService(IJobRepository jobRepository)
    {
        _jobRepository = jobRepository;
    }

    public IEnumerable<Job> GetAll()
    {
        return _jobRepository.GetAll();
    }

    public Job GetById(Guid id)
    {
        return _jobRepository.GetById(id);
    }

    public Task<bool> UpdateTrigger(Guid id, string trigger)
    {
        return _jobRepository.UpdateTrigger(id, trigger);
    }

    public Task ResetAllJobs()
    {
        return _jobRepository.ResetAllJobs();
    }
}